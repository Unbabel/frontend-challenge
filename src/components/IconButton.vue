<template>
  <button class="icon-button" v-on:click="onClick" v-bind:title="title">
    <slot></slot>
  </button>
</template>

<script>
export default {
  name: 'IconButton',
  props: {
    title: String,
    onClick: Function
  }
}
</script>

<style scoped>
button {
  background: none;
  border: none;
  -webkit-appearance: none; 
}
</style>