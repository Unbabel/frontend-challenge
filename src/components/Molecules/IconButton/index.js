import React from 'react'
import PropTypes from 'prop-types'
import styled, { css, keyframes } from 'styled-components'
import { ifProp, prop } from 'styled-tools'
import Icon from '../../Atoms/Icon/'
import Button from '../../Atoms/Button/'

/** *************************************
 *
 * ICON BUTTON Component
 *
 * Variants:
 *
 ************************************** */

const StyledButton = styled(Button)`
  max-width: ${props => props.hasText ? '100%' : '2.5em'};
  width: ${ifProp('hasText', 'auto', '2.5em')};
  padding: ${ifProp('hasText', '0 0.4375em', 0)};
  flex: 0 0 2.5em;
  box-sizing: border-box;
  border:none;
`

const Text = styled.span`
  padding: 0.4375em;
  }
`

const Wrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
`

const StyledIcon = styled(Icon)`
  flex: none;
`

const IconButton = ({ icon, children, ...props }) => {
  const {right, height,
  } = props
  const iconElement = <StyledIcon height={height ? height / 2.5 : undefined} icon={icon} display="none" />
  return (
    <StyledButton hasText={!!children} {...props}>
      <Wrapper>
        {right || iconElement}
        {children
          && <Text className="text" >{children}</Text>
        }
        {right && iconElement}
      </Wrapper>
    </StyledButton>
  )
}


IconButton.propTypes = {
  icon: PropTypes.string.isRequired,
  responsive: PropTypes.bool,
  breakpoint: PropTypes.number,
  right: PropTypes.bool,
  height: PropTypes.number,
  children: PropTypes.node,
}

IconButton.defaultProps = {
  breakpoint: 420,
}

export default IconButton
