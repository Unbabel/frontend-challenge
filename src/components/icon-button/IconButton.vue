<script setup>
	const emit = defineEmits(["onClickCallback"]);

	defineProps({
		icon: {
			type: String,
			required: true,
		},
	});

	const onClickHandler = () => emit("onClickCallback");
</script>

<template>
	<button
		class="icon-button"
		@click="onClickHandler"
		data-test="icon-button"
		aria-label="Icon Button"
	>
		<img :src="icon" alt="button-icon" />
	</button>
</template>

<style scoped>
	.icon-button {
		background-color: transparent;
		border: none;
	}

	.icon-button:hover {
		cursor: pointer;
	}
</style>
