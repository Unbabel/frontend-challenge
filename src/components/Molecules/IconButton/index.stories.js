
import React from 'react'
import { storiesOf } from '@storybook/react'
import IconButton from '.'

storiesOf('Molecules/IconButton', module)
  .add('default', () => (
    <IconButton icon="person" transparent />
  ))
  .add('Person', () => (
    <IconButton icon="person" transparent />
  ))
  .add('Upload', () => (
    <IconButton icon="upload" transparent />
  ))
  .add('Delete', () => (
    <IconButton icon="delete" transparent />
  ))
  .add('Fetch Document', () => (
    <IconButton icon="fetch-document" transparent />
  ))
  .add('Add Row', () => (
    <IconButton icon="add-row" transparent />
  ))
  .add('transparent', () => (
    <IconButton icon="person" transparent>Hello</IconButton>
  ))
  .add('with icon on right', () => (
    <IconButton icon="person" right>Hello</IconButton>
  ))

  .add('without text', () => (
    <IconButton icon="person" />
  ))

  .add('height', () => (
    <IconButton icon="person" height={100}>Hello</IconButton>
  ))
