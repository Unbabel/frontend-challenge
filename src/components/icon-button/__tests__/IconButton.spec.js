import { describe, it, expect, beforeEach } from "vitest";

import { mount } from "@vue/test-utils";

// Components
import IconButton from "../IconButton.vue";

describe("IconButton", () => {
  let wrapper;
  let testProps = { icon: "/src/assets/upload.svg" };

  beforeEach(() => {
    // Arrange
    wrapper = mount(IconButton, {
      propsData: testProps,
    });
  });

  it("matches snapshot", () => {
    // Assert
    expect(wrapper).toMatchSnapshot();
  });

  it("has an image element with source defined", () => {
    // Assert
    const imageEl = wrapper.find("img");
    expect(imageEl.exists()).toBe(true);
    expect(imageEl.attributes("src")).toBe("/src/assets/upload.svg");
  });

  it("should trigger emit on click event", async () => {
    // Act
    await wrapper.trigger("click");

    // Assert
    expect(wrapper.emitted().onClickCallback.length).toBe(1);
  });
});
