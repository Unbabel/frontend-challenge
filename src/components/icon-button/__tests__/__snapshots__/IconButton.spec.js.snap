// Vitest Snapshot v1

exports[`IconButton > matches snapshot 1`] = `
VueWrapper {
  "_emitted": {},
  "_emittedByOrder": [],
  "isFunctionalComponent": undefined,
}
`;
